### PR TITLE
fix: allowing routes with more specificity

### PR DESCRIPTION
**Pull request by jordanstith15**

### CODE REVIEW

File: apps/ifc-example-client/ifc-cli.config.js [hunk 1]
```diff
@@ -16,6 +16,15 @@ module.exports = function (frameRouter) {
...
       },
+      application3: {
+        // We are removing the built-in proxy route example for now, as there are conflicts with vite and
+        // we will be adding improvements to the proxy behavior in the near future
+        url: `http://${hostname}:8080/client-app-2/#/`,
+        assignedRoute: "/app2/test",
+        allow: "camera http://localhost:8080;", // optional
+        sandbox: "allow-presentation allow-modals", // optional
+        defaultTitle: "iframe Application 2 Example", // optional, but needed for accessibility
+      },
```
Comment (jordanstith15) [inline apps/ifc-example-client/ifc-cli.config.js:27]: Added extra app to test the client id and routes with more specificity

File: packages/iframe-coordinator-cli/src/views/IframeEmbed.vue [hunk 2]
```diff
@@ -5,6 +5,8 @@
       <span class="app-route">{{ frameRoute }}</span>
       as
       <span class="frame-url">{{ frameUrl }}</span>
+      with Client Id of
+      <span class="client-id">{{ currentClientId }}</span>
```
Comment (jordanstith15) [inline packages/iframe-coordinator-cli/src/views/IframeEmbed.vue:9]: showing current client id so it's easier to see and test whether the correct client is loaded

File: packages/iframe-coordinator/src/HostRouter.ts [hunk 3]
```diff
@@ -13,9 +13,14 @@ export class HostRouter {
   private _clients: ClientInfo[];
 
   constructor(clients: RoutingMap) {
-    this._clients = Object.keys(clients).map((id) => {
-      return parseRegistration(id, clients[id]);
-    });
+    this._clients = Object.entries(clients)
```
Comment (jordanstith15) [inline packages/iframe-coordinator/src/HostRouter.ts:16]: Noticed that we were doing essentially Object.entries and doing brackets to get array data, which I've seen flagged before on code analyzers.

File: packages/iframe-coordinator/src/HostRouter.ts [hunk 4]
```diff
@@ -13,9 +13,14 @@ export class HostRouter {
...
+      .map(([id, data]) => {
+        return parseRegistration(id, data);
+      })
+      .sort(
+        (a, b) =>
+          b.assignedRoute.split("/").length - a.assignedRoute.split("/").length,
```
Comment (jordanstith15) [inline packages/iframe-coordinator/src/HostRouter.ts:22]: Sorts clients by number of url segments descending

File: packages/iframe-coordinator/src/HostRouter.ts [hunk 5]
```diff
@@ -26,7 +31,7 @@ export class HostRouter {
   public getClientTarget(route: string): ClientTarget | null {
     let clientTarget: ClientTarget | null = null;
 
-    this._clients.forEach((client) => {
+    for (let client of this._clients) {
```
Comment (jordanstith15) [inline packages/iframe-coordinator/src/HostRouter.ts:34]: Noticed this issue as well.
It was continuing to loop after finding a matching client, causing it to seem like it was searching back to front.
Now it breaks the loop after finding a matching client and returns the first match.

File: packages/iframe-coordinator/src/HostRouter.ts (hunk 4 above)
Comment (jordanstith15) [reply packages/iframe-coordinator/src/HostRouter.ts:22]: No clue where the comma came from. I think the linter added it on commit

**Review by MattCheely — CHANGES_REQUESTED** [review]
We should add a new unit test against `getClientTarget` in `HostRoute.spec.ts` that would have failed previously but now succeeds.

**Comment by MattCheely** [comment]
I think this has to be flagged as a breaking change because it can change which app shows in response to a specific URL. Technically it was a bug, but in theory there could be an app out there that accidentally relies on this behavior.

**Review by MattCheely — APPROVED** [review]

**Pull request merged**